### PR TITLE
Added the ability to use non uniform distribution for variations

### PR DESCRIPTION
**Pull request by Guillaume-Caillet**
Also small fix for method 'get_exp' with the 'id' parameter.

### CODE REVIEW

File: flask_google_optimize.py [hunk 1]
```diff
@@ -37,7 +45,7 @@ def get_exp(self, key=None, id=None):
         if key:
             return self._experiments[key]
         if id:
-            return [e for e in self._experiments if e['id'] == id][0]
+            return [e for e in self._experiments if e.id == id][0]
```
Comment (rkz) [inline flask_google_optimize.py:48]: Nice catch :)

File: flask_google_optimize.py [hunk 2]
```diff
@@ -83,13 +92,36 @@ def run(self, experiment_key):
...
+        Return the index of the randomly chosen variation. Choice is by default non uniform and depends on the weight
+        of each variation (you can of course simulate a uniform variation by giving the same weight to each variation
+
+        Implementation algorithm is quite simple : we construct a list of variation indexes. Each index is repeated
+        more or less depending on its weight. And then it's just a matter of using the random.choice function
+        to randomly pick a choice in this weighted list.
+
+        :param variations: A list of {'key':..., 'weight'...}
+        :return: Integer. The index of the chosen variation from variations
+        """
```
Comment (rkz) [inline flask_google_optimize.py:129]: Superbe doc, j'adore !

File: flask_google_optimize.py [hunk 3]
```diff
@@ -48,13 +56,14 @@ class Experiment(object):
     def __init__(self, key, id, variations):
         self.key = key
         self.id = id
-        self.variations = variations  # {var_id: var_key}
+        self.variations = variations  # [{'key': 'small', 'weight': 0.5},...]
```
Comment (rkz) [inline flask_google_optimize.py:59]: Manque une validation du format et de la somme des poids = 1.

**Review by rkz — CHANGES_REQUESTED** [review]

File: flask_google_optimize.py (hunk 3 above)
Comment (Guillaume-Caillet) [reply flask_google_optimize.py:59]: I can add a check on the type of weight and ensuring the total is less than 1.

But having a total strictly equals to 1 is not required for the algorithm, so I'm not sure about this one. Having 2 variations both with a weight of 0.2 and 0.2 will produce the expected results : both variation will get the same probability to be chosen.

**Pull request merged**